kebechet/managers/thoth_advise/thoth_advise.py

```diff
@@ -311,51 +311,43 @@ def run(self, labels: list, analysis_id=None):
 
                 with open(".thoth.yaml", "r") as f:
                     thoth_config = yaml.safe_load(f)
-
-                if thoth_config.get("overlays_dir"):
-                    runtime_environments = [
-                        e["name"] for e in thoth_config["runtime_environments"]
-                    ]
-                else:
-                    runtime_environments = [
-                        thoth_config["runtime_environments"][0]["name"]
-                    ]
-                for e in runtime_environments:
-                    try:
-                        analysis_id = lib.advise_here(
-                            nowait=True,
-                            origin=(f"{self.service_url}/{self.slug}"),
-                            source_type=ThothAdviserIntegrationEnum.KEBECHET,
-                            kebechet_metadata=self.metadata,
-                            runtime_environment_name=e,
-                        )
-                        self._tracking_issue.comment(
-                            STARTED_ADVISE_COMMENT.format(
-                                analysis_id=analysis_id,
-                                env=e,
-                                host=thoth_config["host"],
-                            )
-                        )
-                    except (FileLoadError, ThothConfigurationError) as exc:
-                        if isinstance(exc, FileLoadError):
-                            self._tracking_issue.comment(
-                                f"""Result for {e}: Error advising, no requirements found.
-
-                                If this project does not use requirements.txt or Pipfile then remove thoth-advise
-                                manager from your .thoth.yaml configuration."""
+                if self.runtime_environments:
```

Comment:
This check shouldn't be necessary if the default would be an empty list. If the default is `None` however, you could change the for loop to:
```
for e in self.runtime_environments or []:
```

Here and in update manager

Reply:
ack done

Reply:
I still see the `if` statement in the current PR. I think it can be handled in the `for` as @KPostOffice mention.